plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -332,8 +332,14 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                     }
                 });
 
+        TupleDomain<IcebergColumnHandle> dynamicFilterTupleDomain = dynamicFilter.getCurrentPredicate().transformKeys(IcebergColumnHandle.class::cast);
+        if (dynamicFilterTupleDomain.getDomains().isPresent() && !dynamicFilterTupleDomain.isAll() && !partitionKeys.isEmpty()) {
+            // Filter out eventual partition filters from the dynamic filter because they are irrelevant at this stage
```

Comment:
It's not clear to me why the dynamic filter on partitioning column is irrelevant at this stage. There is no guarantee that the dynamic filter that was applied on the coordinator was "complete", it's possible that the dynamic filter at this stage more selective than the one in coordinator. There was an open issue about it as well https://github.com/trinodb/trino/issues/16299

Reply:
Makes sense to me.

In the test provided with the PR, if it completes only while being on the `IcebergPageSourceProvider`, it is important to actually prune the split.

```
The dynamic filter reduces the range specified for the partition column `date` from `date :: [[2023-01-01, 2024-01-01]]` to `date :: {[2023-01-02]}`
```

> It's not clear to me why the dynamic filter on partitioning column is irrelevant at this stage.

I've missed this use case of the dynamic filter.

cc @Dith3r @sopel39